ui-admin/src/study/participants/datarepo/DatasetList.tsx

```diff
@@ -23,8 +23,12 @@ const datasetColumns = (currentEnvPath: string): ColumnDef<DatasetDetails>[] =>
   id: 'datasetName',
   header: 'Dataset Name',
   accessorKey: 'datasetName',
-  cell: info => <Link to={getDatasetDashboardPath(info.getValue() as unknown as string, currentEnvPath)}
-    className="mx-2">{info.getValue() as unknown as string}</Link>
+  cell: info => {
+    return info.row.original.status !== 'DELETING' ?
+      <Link to={getDatasetDashboardPath(info.getValue() as unknown as string, currentEnvPath)} className="mx-2">
+        {info.getValue() as unknown as string}
+      </Link> : <span className="mx-2">{info.getValue() as unknown as string}</span>
```

Comment:
Oof... `as unknown as string` is unsightly. I haven't tried it yet, but I recently noticed docs about "column helpers" (https://tanstack.com/table/v8/docs/guide/column-defs#column-helpers) that supposedly make types work better. I wonder if that would help in this case.

Reply:
I didn't spend too long on it, but I wasn't able to get the column helpers to work. But I did make this a bit less ugly by accessing the typed fields through the underlying `DatasetDetails` type instead: 8b2a7ebf